### PR TITLE
Image Quality Analysis (IQA) API

### PR DESCRIPTION
**Pull request by clunietp**
### This pullrequest changes

Implements an image quality analysis API

Needed a new PR due to git wrangling.  Continuation of https://github.com/opencv/opencv_contrib/pull/1959

### CODE REVIEW

**Comment by alalek** [comment]
No need to reopen PR or create new branch - just use "--force" option with "git push" command.

File: modules/quality/src/precomp.hpp [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#ifndef OPENCV_QUALITY_PRECOMP_HPP
+#define OPENCV_QUALITY_PRECOMP_HPP
+#endif
```
Comment (alalek) [inline modules/quality/src/precomp.hpp]: Consider including here core module headers at least:
```
#include <opencv2/core.hpp>
```

File: modules/quality/include/opencv2/quality/quality_utils.hpp [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    for (auto& mat : mats)
+        result.emplace_back(expand_mat<R>(mat, TYPE_DEFAULT ));
+
+    return result;
+}
+
+// multiply a cv::Scalar by a scalar
+inline void scalar_multiply(cv::Scalar& what, double val)
+{
+    cv::multiply(what, cv::Scalar{ val, val, val, val }, what);
```
[the remark this answers is not in the document]
Comment (clunietp) [reply modules/quality/include/opencv2/quality/quality_utils.hpp]: Fixed thank you

File: modules/quality/test/test_precomp.hpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        EXPECT_GT(qm.rows, 0);
+        EXPECT_GT(qm.cols, 0);
+    }
+
+    ptr->clear();
+    EXPECT_TRUE(ptr->empty());
+}
+
+template <typename Fn>
+inline void quality_performance_test( const char* name, Fn&& op )
```
[the remark this answers is not in the document]
Comment (clunietp) [reply modules/quality/test/test_precomp.hpp]: Removed internal a/b perf tests

**Comment by clunietp** [comment]
Thank you @alalek !

**Comment by PhilLab** [comment]
Thank you for the new module and contribution 👍 

@alalek While the README.md is much more convenient in Github, the [official documentation](https://docs.opencv.org/master/dc/d20/group__quality.html) completely ignores it. And not only, for this module, also (for example) the [cnn3dobj Readme](https://github.com/opencv/opencv_contrib/blob/master/modules/cnn_3dobj/README.md) is [omitted](https://docs.opencv.org/master/d9/d02/group__cnn__3dobj.html). I am not familiar with the OpenCV doc build toolchain, but in our projects, we have a script which adds ``@addtogroup XYZ`` to the beginning of each README.md and thus it appears in the module overview page

**Comment by clunietp** [comment]
@PhilLab Yes I need to improve the OpenCV docs for this module, thank you for the reminder!  The `@addtogroup` trick at the top of a README.md would be great (if it works here)

**Pull request merged**